File: 2017/4/solve.py
#! /usr/bin/env python
"""Solution to the AoC problem"""

import os


def part1(_input):
    """Part 1 solution implementation"""

    return len([p for p in _input if is_passphrase_valid(p.split(' '))])


def part2(_input):
    """Part 2 solution implementation"""

    return len([p for p in _input if is_passphrase_valid_v2(p.split(' '))])


def is_passphrase_valid(passphrase):
    return len(passphrase) == len(set(passphrase))


def is_passphrase_valid_v2(passphrase):
    sorted_pass = [''.join(sorted(w)) for w in passphrase]
    return len(sorted_pass) == len(set(sorted_pass))


if __name__ == '__main__':
    input_path = os.path.join(
        os.path.dirname(__file__),
        'input')
    _input = open(input_path, 'r').read().strip().split('\n')

    print('Part 1 solution: {}'.format(part1(_input) or 'not solved :('))
    print('Part 2 solution: {}'.format(part2(_input) or 'nope.'))

File: boilerplates/2020/solve.js
const fs = require("fs");
const path = require("path");

const part1 = (input) => {
  return null;
};

const part2 = (input) => {
  return null;
};

const inputPath = path.join(__dirname, "input");
const inputFile = fs
  .readFileSync(inputPath, "utf8")
  .split("\n")
  .filter((a) => a !== "");

const solution1 = part1(inputFile);
if (solution1 !== null) {
  console.log(`  Solution for part 1 is: ${solution1}`);
}

const solution2 = part2(inputFile);
if (solution2 !== null) {
  console.log(`  Solution for part 2 is: ${solution2}`);
}
